### PR TITLE
Task/04 styling

### PR DESCRIPTION
**Pull request by greyzerua**
- [x] Mentor is added to pr
- [x] AC copied to pr description
- [x] Link to deployed version is added to pr https://greyzerua.github.io/fe-react-2024/
- [x] Added layout of Search Bar
- [x] Rewritten all styles with Module Styles
- [x] Implemented dynamic theme change logic
- [x] Implemented logic for fetching user's theme.
- [x] Saved theme to the Local storage.
photo_2024-05-19_23-25-29

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -3,19 +3,29 @@ import { useState } from 'react';
...
 const App = () => {
+    const defaultTheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? ETheme.DARK : ETheme.LIGHT;
     const selectedProductsStorage = JSON.parse(localStorage.getItem(SELECTED_PRODUCTS_KEY) || JSON.stringify({}));
+    const selectedTheme = (localStorage.getItem(THEME_KEY) as ETheme | null) || defaultTheme;
 
     const [currentPage, setCurrentPage] = useState<EAppPage>(EAppPage.ABOUT);
     const [selectedProducts, setSelectedProducts] = useState<SelectedProducts>(selectedProductsStorage);
+    const [currentTheme, setCurrentTheme] = useState<ETheme>(selectedTheme);
+
+    const onThemeChange = (theme: ETheme) => {
```
Comment (vitalii-mel-work) [inline src/App.tsx]: create change theme components and mov logic 

in app only get from lockalstorage theme type

File: src/App.tsx [hunk 2]
```diff
@@ -3,19 +3,29 @@ import { useState } from 'react';
...
+import { ETheme, THEME_CLASSES } from './constants/themes';
 import { AboutPage } from './pages/about-page';
 import { ProductsPage } from './pages/products-page';
 import type { AddSelectedProduct, SelectedProducts } from './types/selected-products';
 
-import './App.css';
+import styles from './App.module.css';
 
 const SELECTED_PRODUCTS_KEY = 'selected-products-key';
+const THEME_KEY = 'theme-key';
```
Comment (vitalii-mel-work) [inline src/App.tsx]: 2keys into config file

**Review by vitalii-mel-work — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (greyzerua) [reply src/App.tsx]: fixed

File: src/App.tsx (hunk 2 above)
Comment (greyzerua) [reply src/App.tsx]: fixed

**Pull request merged**